language/move-core/types/src/u256.rs

```diff
@@ -1,18 +1,139 @@
 // Copyright (c) The Move Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use std::fmt;
+use std::{
+    fmt,
+    ops::{BitAnd, BitOr, BitXor, Shl, Shr},
+};
 
 // This U256 impl was chosen for now but we are open to changing it as needed
 use primitive_types::U256;
 use serde::{Deserialize, Serialize};
 
 #[allow(non_camel_case_types)]
-#[derive(Clone, Deserialize, Serialize, Debug, PartialEq, Eq, Hash, Copy)]
+#[derive(Clone, Deserialize, Serialize, Debug, PartialEq, Eq, Hash, Copy, PartialOrd, Ord)]
 pub struct u256(U256);
```

Comment:
I kinda feel like we should rename the struct to U256 with a capital U. By rust convention, type names starting with a lower letter is reserved for primitive types. It might become problematic if let's say, rust adds its own u256 type one day.

Reply:
Yes I initially used the uppercase U256 but there were conflicts due to `U256` also having to be the Move type name in various parts of the code